docs/privatelink.md

```diff
@@ -0,0 +1,276 @@
+the cluster's API.
+
+### Configure the primary GCP credentailsSecretRef
+
+The Link Actuator needs a GCP Service Account in order to manage the Endpoint.
+The credentials for this service account should be stored in a secret in the
+Hive namespace. This secret can then be configured via the
+`credentailsSecretRef` parameter in the hiveconfig.
+
+```yaml
+## hiveconfig
+spec:
+  privateLink:
+    gcp:
+      ## credentialsSecretRef points to a secret with permissions to create
+      ## resources in account where the inventory of VPCs exist.
+      credentialsSecretRef:
+        name: < link-account-credentials-secret-name >
+```
+
+This service account requires the following permissions:
+
+- Endpoint Address
+    - compute.addresses.create
+    - compute.addresses.createInternal
+    - compute.addresses.delete
+    - compute.addresses.deleteInternal
+    - compute.addresses.get
+    - compute.addresses.list
+    - compute.instances.update
+    - compute.regionOperations.get
+    - compute.subnetworks.get
+- Endpoint
+    - compute.addresses.use
+    - compute.forwardingRules.create
+    - compute.forwardingRules.delete
+    - compute.forwardingRules.get
+    - compute.forwardingRules.pscCreate
+    - compute.forwardingRules.pscDelete
+    - compute.networks.use
+    - compute.regionOperations.get
+    - compute.subnetworks.use
+    - servicedirectory.namespaces.create
+    - servicedirectory.services.create
+    - servicedirectory.services.delete
+
+
+### Configure the GCP endpointVPCInventory
+
+The Link Actuator needs to know where to create the cluster's Endpoint. This
+can be configured in the `spec.privateLink.gcp.endpointVPCInventory` parameter
+in the hiveconfig. When creating the endpoint, the actuator will choose the
+subnet from this list that is in the same region and has the least number of
+addresses provisioned in the cloud provider.
+
+1. Create a network for the PrivateLink Controller to use when creating
+endpoints.
+
+1. Create one or more subnets in the network for each region that private
+clusters will be created in.
+
+1. Make sure all the Hive environments (Hive VPCs) have network reachability to
+these subnets using peering, transit gateways, VPNs, etc.
+
+1. Update the HiveConfig to enable these subnets.
+
+```yaml
+## hiveconfig
+spec:
+  privateLink:
+    gcp:
+      endpointVPCInventory:
+      - network: network-1
+        subnets:
+        - region: us-east1
+          subnet: subnet1
+        - region: us-east2
+          subnet: subnet2
+```
+
+## AWS Deploying a PrivateLink cluster
+
+Support for AWS private clusters is still managed by the origional [AWS
+PrivateLink Controller](awsprivatelink.md). The goal is to eventually merge its
+functionality into this PrivateLink Controller.
+
+## GCP Deploying a PrivateLink cluster
+
+Once Hive has been configured to support PrivateLink GCP clusters, you can
+deploy a cluster by setting `privateServiceConnect.enabled` to true. This is
```

Comment:
```suggestion
deploy a cluster by setting `privateServiceConnect.enabled` to true on the clusterDeployment. This is
```

(I know this is made obvious below, but I think it flows a bit better this way since we've previously been talking almost exclusively about the hiveconfig.)

Reply:
[Fixed](https://github.com/openshift/hive/compare/19933d189d2166ec5fd4733e83297f6d742652b8..3df83c8128307810f05ed8391f1c67eaa57c0ff8)